stonesoup/models/measurement/nonlinear.py

```diff
@@ -517,7 +552,322 @@ def function(self, state, noise=False, **kwargs):
 
         return StateVector([[Elevation(theta)], [Bearing(phi)]]) + noise
 
-    def rvs(self, num_samples=1, **kwargs):
+    def rvs(self, num_samples=1, **kwargs) -> np.ndarray:
         out = super().rvs(num_samples, **kwargs)
         out = np.array([[Elevation(0.)], [Bearing(0.)]]) + out
         return out
+
+
+class CartesianToBearingRangeRate(NonLinearGaussianMeasurement):
+    r"""This is a class implementation of a time-invariant measurement model, \
+    where measurements are assumed to be received in the form of bearing \
+    (:math:`\phi`), range (:math:`r`) and range-rate (:math:`\dot{r}`),
+    with Gaussian noise in each dimension.
+
+    The model is described by the following equations:
+
+    .. math::
+
+      \vec{y}_t = h(\vec{x}_t, \vec{v}_t)
+
+    where:
+
+    * :math:`\vec{y}_t` is a measurement vector of the form:
+
+    .. math::
+
+      \vec{y}_t = \begin{bmatrix}
+                \phi \\
+                r \\
+                \dot{r}
+            \end{bmatrix}
+
+    * :math:`h` is a non-linear model function of the form:
+
+    .. math::
+
+      h(\vec{x}_t,\vec{v}_t) = \begin{bmatrix}
+                atan2(\mathcal{y},\mathcal{x}) \\
+                \sqrt{\mathcal{x}^2 + \mathcal{y}^2} \\
+                (x\dot{x} + y\dot{y})/\sqrt{x^2 + y^2}
+                \end{bmatrix} + \vec{v}_t
+
+    * :math:`\vec{v}_t` is Gaussian distributed with covariance\
+     :math:`R`, i.e.:
+
+    .. math::
+
+      \vec{v}_t \sim \mathcal{N}(0,R)
+
+    .. math::
+
+      R = \begin{bmatrix}
+            \sigma_{\phi}^2 & 0 & 0\\
+            0 & \sigma_{r}^2 & 0 \\
+            0 & 0 & \sigma_{\dot{r}}^2
+            \end{bmatrix}
+
+    The :py:attr:`mapping` property of the model is a 3 element vector, \
+    whose first (i.e. :py:attr:`mapping[0]`), second (i.e. \
+    :py:attr:`mapping[1]`) and third (i.e. :py:attr:`mapping[2]`) elements \
+    contain the state index of the :math:`x`, :math:`y` and :math:`z`  \
+    coordinates, respectively.
+
+    Note
+    ----
+    This class implementation assuming at 3D cartesian space, it therefore\
+     expects a 6D state space.
+    """
+
+    translation_offset = Property(
+        StateVector, default=None,
+        doc="A 3x1 array specifying the origin offset in terms of :math:`x,y` coordinates.")
+    velocity_mapping = Property(
+        np.array, default=(1, 3, 5),
+        doc="Mapping to the targets velocity within its state space")
+    velocity = Property(
+        StateVector, default=None,
+        doc="A 3x1 array specifying the sensor velocity in terms of :math:`x,y` \
+        coordinates.")
+
+    def __init__(self, *args, **kwargs):
+        """
+        Ensure that the translation offset is initiated
+        """
+        super().__init__(*args, **kwargs)
```

Comment:
Initially I started with it being 2D and moved to make it 3D as it was causing some issues (i.e. I did it badly!). Your logic above would resolve those issues so I'll have a quick look now and see if i can re-issue

Reply:
Actually, it would make sense for all these models to be capable in both 2/3d cases. Probably a separate pull request as there are a fair few little changes and tests which would need to be generated.